L1Trigger/Phase2L1ParticleFlow/test/testOutputFiles.cpp

```diff
@@ -0,0 +1,611 @@
+// STL includes
+#include <cstdio>
+#include <iostream>
+#include <iomanip>
+#include <string>
+#include <stdexcept>
+#include <sstream>
+#include <fstream>
+#include <limits>
+#include <vector>
+#include <map>
+#include <utility>
+#include <math.h>
+#include <bitset>
+
+// ROOT includes
+#include "TROOT.h"
+#include "TSystem.h"
+#include "TFile.h"
+#include "TTree.h"
+#include "TLorentzVector.h"
+
+// CMSSW includes
+#include "FWCore/FWLite/interface/FWLiteEnabler.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/FWLite/interface/Handle.h"
+#include "DataFormats/FWLite/interface/Run.h"
+#include "DataFormats/FWLite/interface/LuminosityBlock.h"
+#include "DataFormats/FWLite/interface/Event.h"
+#include "DataFormats/L1TParticleFlow/interface/PFTrack.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/DiscretePFInputsIO.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/Region.h"
+
+#define NTEST 64
+#define REPORT_EVERY_N 50
+#define NTRACKS_PER_SECTOR 110
+#define NBITS_PER_TRACK 96
+static std::vector<l1tpf_impl::Region> regions_;
+
+typedef l1tpf_impl::InputRegion Region;
+typedef std::pair<int, int> SectorTrackIndex;
+typedef std::map<SectorTrackIndex, TLorentzVector *> TrackMap;
+
+struct Event {
+  uint32_t run, lumi;
+  uint64_t event;
+  float z0, genZ0;
+  std::vector<float> puGlobals;  // [float] alphaCMed, alphaCRms, alphaFMed, alphaFRms
+  std::vector<Region> regions;
+
+  Event() : run(0), lumi(0), event(0), z0(0.), regions() {}
+  bool readFromFile(FILE *fRegionDump) {
+    if (!fread(&run, sizeof(uint32_t), 1, fRegionDump))
+      return false;
+    fread(&lumi, sizeof(uint32_t), 1, fRegionDump);
+    fread(&event, sizeof(uint64_t), 1, fRegionDump);
+    l1tpf_impl::readManyFromFile(regions, fRegionDump);
+    fread(&z0, sizeof(float), 1, fRegionDump);
+    fread(&genZ0, sizeof(float), 1, fRegionDump);
+    l1tpf_impl::readManyFromFile(puGlobals, fRegionDump);
+    return true;
+  }
+};
+
+TLorentzVector *makeTLorentzVectorPtEtaPhiE(float pt, float eta, float phi, float e) {
+  TLorentzVector *v = new TLorentzVector();
```

Comment:
no need for a heap allocation here

Reply:
This is a test not used anywhere in production.

Reply:
anyway, moved the TLorentzVectors to be used by value instead of pointer everywhere.